### PR TITLE
parametric: enable python otel env telemetry tests

### PR DESCRIPTION
**Pull request by mabdinur**
## Motivation

Enable Instrumentation Telemetry test for OpenTelemetry environment variable support.

## Changes

- Enable tests for python
- Update tests to verify metric tags that are not set by ruby or golang. 

### TODO

- The go, ruby, and nodejs tracers submit telemetry under different namespaces and set different tags for some otel enviornments. We will need to address these consistencies.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: tests/parametric/test_telemetry.py [hunk 1]
```diff
@@ -308,41 +308,52 @@ def test_telemetry_otel_env_hiding(self, library_env, test_agent, test_library):
...
-
-        for expected in expected_tags:
-            assert any(all(tag in metric["tags"] for tag in expected) for metric in otelHiding)
-
-        for metric in otelHiding:
-            assert metric["points"][0][1] == 1
+        if context.library == "python":
+            otelsampler_config = "otel_traces_sampler"
+        else:
+            otelsampler_config = "otel_traces_sampler_args"
```
Comment (mabdinur) [inline tests/parametric/test_telemetry.py:326]:
```suggestion
            otelsampler_config = "otel_traces_sampler_arg"
```

File: tests/parametric/test_telemetry.py [hunk 2]
```diff
@@ -373,32 +385,43 @@ def test_telemetry_otel_env_invalid(self, library_env, test_agent, test_library)
...
+            ddlog_config = "dd_trace_log_level"
+        elif context.library == "python":
+            ddlog_config = "dd_trace_debug"
+        else:
+            ddlog_config = "dd_log_level"
+
+        if context.library == "python":
+            otelsampler_config = "otel_traces_sampler"
+        else:
+            otelsampler_config = "otel_traces_sampler_args"
```
Comment (mabdinur) [inline tests/parametric/test_telemetry.py:402]:
```suggestion
            otelsampler_config = "otel_traces_sampler_arg"
```

File: tests/parametric/test_telemetry.py [hunk 3]
```diff
@@ -373,32 +385,43 @@ def test_telemetry_otel_env_invalid(self, library_env, test_agent, test_library)
...
+            otelsampler_config = "otel_traces_sampler"
+        else:
+            otelsampler_config = "otel_traces_sampler_arg"
+
+        for dd_config, otel_config in [
+            ["dd_trace_propagation_style", "otel_propagators"],
+            ["dd_trace_sample_rate", "otel_traces_sampler"],
+            ["dd_trace_enabled", "otel_traces_exporter"],
+            ["dd_runtime_metrics_enabled", "otel_metrics_exporter"],
+            ["dd_tags", "otel_resource_attributes"],
```
Comment (mabdinur) [inline tests/parametric/test_telemetry.py:409]:
```suggestion
            # ["dd_tags", "otel_resource_attributes"], not supported by nodejs
```

**Review by robertomonteromiguel — APPROVED** [review]

File: tests/parametric/test_telemetry.py [hunk 4]
```diff
@@ -373,32 +385,43 @@ def test_telemetry_otel_env_invalid(self, library_env, test_agent, test_library)
...
+            otelsampler_config = "otel_traces_sampler"
+        else:
+            otelsampler_config = "otel_traces_sampler_arg"
+
+        for dd_config, otel_config in [
+            ["dd_trace_propagation_style", "otel_propagators"],
+            ["dd_trace_sample_rate", "otel_traces_sampler"],
+            ["dd_trace_enabled", "otel_traces_exporter"],
+            ["dd_runtime_metrics_enabled", "otel_metrics_exporter"],
+            # ["dd_tags", "otel_resource_attributes"], not supported by nodejs
```
Comment (mabdinur) [inline tests/parametric/test_telemetry.py:409]:
```suggestion
            # ["dd_tags", "otel_resource_attributes"],
```

File: tests/parametric/test_telemetry.py [hunk 5]
```diff
@@ -308,41 +308,52 @@ def test_telemetry_otel_env_hiding(self, library_env, test_agent, test_library):
...
+                    False
+                ), f"Could not find a metric with {dd_config} and {otel_config} in otelHiding metrics: {otelHiding}"
 
     @missing_feature(context.library == "dotnet", reason="Not implemented")
     @missing_feature(context.library == "java", reason="Not implemented")
     @missing_feature(context.library == "ruby", reason="Not implemented")
     @missing_feature(context.library == "php", reason="Not implemented")
     @missing_feature(context.library == "cpp", reason="Not implemented")
-    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(context.library < "python@2.18.0.dev", reason="Not implemented")
```
Comment (mabdinur) [inline tests/parametric/test_telemetry.py:356]:
```suggestion
    @missing_feature(context.library == "does not collect invalid metrics for otel_resource_attributes", reason="nodejs")
```

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
lint failures

File: tests/parametric/test_telemetry.py [hunk 6]
```diff
@@ -308,41 +308,53 @@ def test_telemetry_otel_env_hiding(self, library_env, test_agent, test_library):
...
+                ), f"Could not find a metric with {dd_config} and {otel_config} in otelHiding metrics: {otelHiding}"
 
     @missing_feature(context.library == "dotnet", reason="Not implemented")
     @missing_feature(context.library == "java", reason="Not implemented")
     @missing_feature(context.library == "ruby", reason="Not implemented")
     @missing_feature(context.library == "php", reason="Not implemented")
     @missing_feature(context.library == "cpp", reason="Not implemented")
-    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(
+        context.library == "does not collect env.invalid metrics for otel_resource_attributes", reason="nodejs"
```
Comment (mabdinur) [inline tests/parametric/test_telemetry.py:356]:
```suggestion
        context.library == "nodejs", reason="does not collect env.invalid metrics for otel_resource_attributes"
```

File: tests/parametric/test_telemetry.py [hunk 7]
```diff
@@ -308,41 +308,53 @@ def test_telemetry_otel_env_hiding(self, library_env, test_agent, test_library):
...
+                ), f"Could not find a metric with {dd_config} and {otel_config} in otelHiding metrics: {otelHiding}"
 
     @missing_feature(context.library == "dotnet", reason="Not implemented")
     @missing_feature(context.library == "java", reason="Not implemented")
     @missing_feature(context.library == "ruby", reason="Not implemented")
     @missing_feature(context.library == "php", reason="Not implemented")
     @missing_feature(context.library == "cpp", reason="Not implemented")
-    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(
+        context.library == "nodejs", reason="does not collect env.invalid metrics for otel_resource_attributes"
```
Comment (mabdinur) [inline tests/parametric/test_telemetry.py:356]:
```suggestion
        context.library == "nodejs", reason="does not collect otel_env.invalid metrics for otel_resource_attributes"
```

File: tests/parametric/test_telemetry.py [hunk 8]
```diff
@@ -308,41 +308,53 @@ def test_telemetry_otel_env_hiding(self, library_env, test_agent, test_library):
...
+            else:
+                assert (
+                    False
+                ), f"Could not find a metric with {dd_config} and {otel_config} in otelHiding metrics: {otelHiding}"
 
     @missing_feature(context.library == "dotnet", reason="Not implemented")
     @missing_feature(context.library == "java", reason="Not implemented")
     @missing_feature(context.library == "ruby", reason="Not implemented")
     @missing_feature(context.library == "php", reason="Not implemented")
     @missing_feature(context.library == "cpp", reason="Not implemented")
```
Comment (mabdinur) [inline tests/parametric/test_telemetry.py:354]:
```suggestion
    @missing_feature(context.library == "cpp", reason="Not implemented")
    @missing_feature(context.library < "python@2.18.0.dev", reason="Not implemented")
```

**Review by cbeauchesne — APPROVED** [review]

**Review by robertomonteromiguel — APPROVED** [review]

**Pull request merged**